src/gasPriceOracle/oracle.ts

```diff
@@ -53,6 +55,21 @@ export async function getGasPriceEstimate(
     `Require 1.0 < priority fee multiplier (${priorityFeeMultiplier}) <= 5.0 for a total gas multiplier within [+1.0, +5.0]`
   );
 
+  const isEthersProvider = provider instanceof providers.Provider;
+  // Exit here if we need to estimate on Solana.
+  if (!isEthersProvider) {
```

Comment:
I wonder if we should make `chainId` a mandatory field in `GasPriceEstimateOptions`, because we can't resolve the Solana "chain Id" via an RPC call, so this won't scale beyond one non-evm/svm chain.

Reply:
I also agree. It's annoying that the default value is {} since I expect that we pass in an empty set of options quite often. I at least made a change to use the opts to get the chain Id if applicable before assuming `CHAIN_IDs.SOLANA` https://github.com/across-protocol/sdk/pull/980/commits/61b885440f23df0eb26b7df511315a4eabe9b1e7